src/node_api.cc

```diff
@@ -309,8 +311,12 @@ class ThreadSafeFunction : public node::AsyncResource {
                                "ERR_NAPI_TSFN_STOP_IDLE_LOOP",
                                "Failed to stop the idle loop") == napi_ok);
       } else {
-        v8::Local<v8::Function> js_cb =
-            v8::Local<v8::Function>::New(env->isolate, ref);
+        v8::Local<v8::Value> js_cb;
+        if (ref.IsEmpty()) {
+          js_cb = v8::Undefined(env->isolate);
```

Comment:
How does this work? Won't this be rejected by the CHECK_TO_FUNCTION check in napi_call_function()? That's what call_js_cb() ends up calling, doesn't it?

Reply:
I agree. It would be better to modify `CallJs` to only call `napi_call_function()` if `cb` is not `nullptr`, and only set `js_cb` here `if (!ref.IsEmpty())`.

Reply:
Yet the `js_cb` would be `CHECK_TO_FUNCTION` at the `napi_create_threadsafe_function` if fallbacking to `CallJs`, in the case `CallJs` would not receive an undefined `js_cb`.